examples/medplum-scheduling-demo/src/components/SetAvailability.tsx

```diff
@@ -0,0 +1,155 @@
+import { Modal } from '@mantine/core';
+import { showNotification } from '@mantine/notifications';
+import { createReference, getAllQuestionnaireAnswers, normalizeErrorString } from '@medplum/core';
+import {
+  Coding,
+  Questionnaire,
+  QuestionnaireItemAnswerOption,
+  QuestionnaireResponse,
+  Schedule,
+} from '@medplum/fhirtypes';
+import { Loading, QuestionnaireForm, useMedplum } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useContext } from 'react';
+import { ScheduleContext } from '../Schedule.context';
+import { SetAvailabilityEvent } from '../bots/core/set-availability';
+
+interface SetAvailabilityProps {
+  readonly opened: boolean;
+  readonly handlers: {
+    readonly open: () => void;
+    readonly close: () => void;
+    readonly toggle: () => void;
+  };
+}
+
+export function SetAvailability(props: SetAvailabilityProps): JSX.Element {
+  const { opened, handlers } = props;
+
+  const medplum = useMedplum();
+
+  const { schedule } = useContext(ScheduleContext);
+
+  if (!schedule) {
+    return <Loading />;
+  }
+
+  async function handleQuestionnaireSubmit(formData: QuestionnaireResponse): Promise<void> {
+    const answers = getAllQuestionnaireAnswers(formData);
+
+    try {
+      // Call 'set-availability' bot to create slots
+      const input: SetAvailabilityEvent = {
+        schedule: createReference(schedule as Schedule),
+        startDate: answers['start-date'][0].valueDate as string,
+        endDate: answers['end-date'][0].valueDate as string,
+        startTime: answers['start-time'][0].valueTime as string,
+        endTime: answers['end-time'][0].valueTime as string,
+        duration: answers['duration'][0].valueInteger as number,
+        daysOfWeek: answers['days-of-week'].map(
+          (d: QuestionnaireItemAnswerOption) => (d.valueCoding as Coding).code as string
+        ),
+      };
+      await medplum.executeBot({ system: 'http://example.com', value: 'set-availability' }, input);
+
+      showNotification({
+        icon: <IconCircleCheck />,
+        title: 'Success',
+        message: 'Slots created',
+      });
+    } catch (err) {
+      showNotification({
+        color: 'red',
+        icon: <IconCircleOff />,
+        title: 'Error',
+        message: normalizeErrorString(err),
+      });
+    }
+
+    handlers.close();
+  }
+
+  return (
+    <Modal opened={opened} onClose={handlers.close}>
+      <QuestionnaireForm
+        questionnaire={setAvailabilityQuestionnaire}
+        subject={createReference(schedule)}
+        onSubmit={handleQuestionnaireSubmit}
+      />
+    </Modal>
+  );
+}
+
+const setAvailabilityQuestionnaire: Questionnaire = {
```

Comment:
Please let me know if there are any copy improvements to make.

![image](https://github.com/user-attachments/assets/cc783987-a2a3-4296-9897-486ea90ff8a6)

Reply:
-  "Start date"
- "End date"
- "Start time (each day)"
- "End time (each day)"
- "Duration of each slot (minutes)"
- "Days of Week"

Reply:
Done